cloudinit/crypt.py

```diff
@@ -0,0 +1,36 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+from contextlib import suppress
+
+
+def _not_available(secret: str, salt: str):
+    """Raise when called so that importing this module doesn't throw
+    ImportError when ds_detect() returns false. In this case, crypt
+    and passlib are not needed.
+    """
+    raise ImportError("crypt/passlib not found, missing dependency")
+
+
+encrypt_pass = _not_available
+_passlib_crypt = _not_available
+_deprecated_crypt = _not_available
+
+
+# lowest priority: external python dependency with an uncertain future
+with suppress(ImportError, AttributeError):
+    import passlib.hash
+
+    def _passlib_crypt(secret: str, salt: str):
+        """crypt.crypt() interface based on passlib"""
+        return passlib.hash.sha512_crypt.hash(secret, salt=salt, rounds=5000)
```

Comment:
Doesn't BSD require `bcrypt` rather than `sha512_crypt`?

Also, this method of calling `hash` is [deprecated](https://passlib.readthedocs.io/en/stable/lib/passlib.ifc.html#passlib.ifc.PasswordHash.hash). I think it should instead look something like
```python
passlib.hash.sha512_crypt.using(salt=salt, rounds=5000).hash(secret)
```